File: backend/app/schemas/calendar.py
from datetime import date
from typing import List, Optional
from pydantic import BaseModel
from app.schemas.user import UserRead


class MemberDayStatus(BaseModel):
    day: date
    status: str  # present (🟩), absent (🟥), freeze (❄️), future (⬜)
    pages_read: Optional[int] = None
    note: Optional[str] = None


class MemberCalendarGrid(BaseModel):
    user: UserRead
    days: List[MemberDayStatus]


class MonthCalendarResponse(BaseModel):
    group_id: str
    month: str  # YYYY-MM
    members: List[MemberCalendarGrid]

File: frontend/src/api/calendar.ts
import { apiClient } from './client';
import { User } from './auth';

export interface MemberDayStatus {
  day: string;
  status: 'present' | 'absent' | 'freeze' | 'future' | 'not_joined';
  pages_read?: number;
  note?: string;
}

export interface MemberCalendarGrid {
  user: User;
  days: MemberDayStatus[];
}

export interface MonthCalendarResponse {
  group_id: string;
  month: string;
  members: MemberCalendarGrid[];
}

export const getGroupCalendar = async (groupId: string, monthStr?: string): Promise<MonthCalendarResponse> => {
  const url = monthStr ? `/groups/${groupId}/calendar?month=${monthStr}` : `/groups/${groupId}/calendar`;
  const response = await apiClient.get<MonthCalendarResponse>(url);
  return response.data;
};
